### PR TITLE
Python launch scripts

### PR DESCRIPTION
**Pull request by cmnrd**
This PR implements the solution proposed in https://github.com/lf-lang/lingua-franca/issues/1458#issuecomment-1640218120. Instead of printing a Python command to run, we generate a shell or batch script in `bin/` which can be executed to run the Python program. This ensures that the executing Python runtime is the same as the one we linked against during compilation.

### CODE REVIEW

File: core/src/main/java/org/lflang/generator/python/PythonGenerator.java [hunk 1]
```diff
@@ -578,7 +573,7 @@ private static String setUpMainTarget(
             add_subdirectory(core)
             set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR})
             set(LF_MAIN_TARGET <pyModuleName>)
-            find_package(Python 3.7.0...<3.11.0 COMPONENTS Interpreter Development)
+            find_package(Python 3.10.0...<3.11.0 REQUIRED COMPONENTS Interpreter Development)
```
Comment (edwardalee) [inline core/src/main/java/org/lflang/generator/python/PythonGenerator.java:576]: Should this be <3.12.0?

**Review by edwardalee — APPROVED** [review]
Great!!  Thanks for fixing this!

File: core/src/main/java/org/lflang/generator/python/PythonGenerator.java (hunk 1 above)
Comment (cmnrd) [reply core/src/main/java/org/lflang/generator/python/PythonGenerator.java:576]: No, this does specifically not add Python 3.11 support. For this we should finalize https://github.com/lf-lang/lingua-franca/pull/1902.

Comment (cmnrd) [reply core/src/main/java/org/lflang/generator/python/PythonGenerator.java:576]: I tried what happens if we just enable Python 3.11 on top of this branch in https://github.com/lf-lang/lingua-franca/pull/1915, but we are getting segfaults. So we still need the fixes in https://github.com/lf-lang/lingua-franca/pull/1902.

**Comment by cmnrd** [comment]
For some reason the Python CI now runs much longer. Has anyone an idea why that could be?

File: core/src/main/java/org/lflang/federated/launcher/PyBuildConfig.java [hunk 2]
```diff
@@ -12,26 +12,12 @@ public PyBuildConfig(
...
-  public String localExecuteCommand() {
-    return "python3 "
-        + fileConfig.getSrcGenPath()
-        + "/"
-        + federate.name
-        + "/"
-        + federate.name
-        + ".py -i $FEDERATION_ID";
+  public String remoteExecuteCommand() {
+    return "bin/" + fileConfig.name + "_" + federate.name + " -i '$FEDERATION_ID'";
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/federated/launcher/PyBuildConfig.java:16]: The slash here doesn't look platform independent, but the code that this replaces also featured slashes...

**Review by lhstrh — APPROVED** [review]

File: core/src/main/java/org/lflang/federated/launcher/PyBuildConfig.java (hunk 2 above)
Comment (lhstrh) [reply core/src/main/java/org/lflang/federated/launcher/PyBuildConfig.java:16]: Created #1917.

**Pull request merged**